packages/server/src/api/controllers/record.js

```diff
@@ -12,6 +12,41 @@ validateJs.extend(validateJs.validators.datetime, {
   },
 })
 
+exports.patch = async function(ctx) {
+  const db = new CouchDB(ctx.user.instanceId)
+  const record = await db.get(ctx.params.id)
+  const model = await db.get(record.modelId)
+  const patchfields = ctx.request.body
+
+  for (let key in patchfields) {
+    if (!model.schema[key]) continue
+    record[key] = patchfields[key]
+  }
+  coerceFieldsToCorrectType(record, model)
```

Comment:
@shogunpurple this method gets run on patch and save... commented below. 

Basically, if someone submits something like 

`{ myBoolField: "true" }`

Then it will get converted to ` { myBoolField: true } `, so that it passes validation. I checked the validatejs docs, to see if it would do it for us, but didn't see anything

Reply:
Ok that makes sense, but we should probably do this in the client If possible when we render mustache templates - it’s not common to do this kind of thing on the server side.

Could we add rules to the mustache render rules to account for this?

Reply:
have moved this into the client.

We are now including the type as part of the action's definition, which gets built into the client.

The client then runs this same code - parsing strings. Now includes datetimes as well